### PR TITLE
make sure the chart fits the onboard requirements

### PR DESCRIPTION
**Pull request by ianzhang366**
the overall dry-run yaml would be:
```
NAME: hypershift-addon
LAST DEPLOYED: Tue Mar  1 12:22:28 2022
NAMESPACE: default
STATUS: pending-install
REVISION: 1
TEST SUITE: None
HOOKS:
MANIFEST:
---
# Source: hypershift-addon-chart/templates/hypershift-addon-manager-serviceaccount.yaml
kind: ServiceAccount
apiVersion: v1
metadata:
  name: hypershift-addon-manager-sa
  namespace: default
---
# Source: hypershift-addon-chart/templates/hypershiftDeployment_serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: hypershift-deployment
  namespace: default
---
# Source: hypershift-addon-chart/templates/hypershiftDeployment_clusterrole.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: open-cluster-management:hypershift-addon-chart:hypershift-deployment-controller
rules:
- apiGroups:
  - ""
  resources:
  - configmaps
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - secrets
  verbs:
  - create
  - deletecollection
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - cluster.open-cluster-management.io
  resources:
  - hypershiftdeployments
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - cluster.open-cluster-management.io
  resources:
  - hypershiftdeployments/finalizers
  verbs:
  - update
- apiGroups:
  - cluster.open-cluster-management.io
  resources:
  - hypershiftdeployments/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - hypershift.openshift.io
  resources:
  - hostedclusters
  - nodepools
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - work.open-cluster-management.io
  resources:
  - manifestworks
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
---
# Source: hypershift-addon-chart/templates/hypershift-addon-manager-clusterrolebinding.yaml
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: open-cluster-management:hypershift-addon-chart:hypershift-addon-manager
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
  - kind: ServiceAccount
    name: hypershift-addon-manager-sa
    namespace: open-cluster-management
---
# Source: hypershift-addon-chart/templates/hypershiftDeployment_clusterrolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: open-cluster-management:hypershift-addon-chart:hypershift-deployment-controller
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: open-cluster-management:hypershift-addon-chart:hypershift-deployment-controller
subjects:
- kind: ServiceAccount
  name: hypershift-deployment
  namespace: default
---
# Source: hypershift-addon-chart/templates/hypershiftDeployment_leader_election_role.yaml
# permissions to do leader election.
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: open-cluster-management:hypershift-addon-chart:hypershiftDeployment-leader-election
  namespace: default
rules:
- apiGroups:
  - ""
  resources:
  - configmaps
  verbs:
  - get
  - list
  - watch
  - create
  - update
  - patch
  - delete
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - get
  - list
  - watch
  - create
  - update
  - patch
  - delete
- apiGroups:
  - ""
  resources:
  - events
  verbs:
  - create
  - patch
---
# Source: hypershift-addon-chart/templates/hypershiftDeployment_leader_election_role_binding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: open-cluster-management:hypershift-addon-chart:hypershiftDeployment-leader-election
  namespace: default
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: open-cluster-management:hypershift-addon-chart:hypershiftDeployment-leader-election
subjects:
- kind: ServiceAccount
  name: hypershift-deployment
  namespace: default
---
# Source: hypershift-addon-chart/templates/hypershift-addon-manager-deployment.yaml
kind: Deployment
apiVersion: apps/v1
metadata:
  name: hypershift-addon-manager
  namespace: default
  labels:
    app: hypershift-addon-manager
spec:
  replicas: 1
  selector:
    matchLabels:
      app: hypershift-addon-manager
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: hypershift-addon-manager
        ocm-antiaffinity-selector: hypershift-addon-manager
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kubernetes.io/arch
                operator: In
                values:
                - amd64
                - ppc64le
                - s390x
                - arm64
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - weight: 70
            podAffinityTerm:
              topologyKey: topology.kubernetes.io/zone
              labelSelector:
                matchExpressions:
                - key: ocm-antiaffinity-selector
                  operator: In
                  values:
                  - hypershift-addon-manager
          - weight: 35
            podAffinityTerm:
              topologyKey: kubernetes.io/hostname
              labelSelector:
                matchExpressions:
                - key: ocm-antiaffinity-selector
                  operator: In
                  values:
                  - hypershift-addon-manager
      restartPolicy: Always
      serviceAccountName: hypershift-addon-manager-sa
      volumes:
      - emptyDir:
          medium: Memory
        name: tmp-vol
      hostNetwork: false
      hostPID: false
      hostIPC: false
      securityContext:
        runAsNonRoot: true
        runAsUser: 1001
      terminationGracePeriodSeconds: 30
      containers:
        - name: hypershift-addon-manager
          image: quay.io/ianzhang366/hypershift-addon-operator:latest
          volumeMounts:
          - mountPath: /tmp
            name: tmp-vol
          args:
            - ./hypershift-addon
            - manager
            - '--namespace=default'
          env:
            - name: HYPERSHIFT_OPERATOR_IMAGE_NAME
              value: 'quay.io/hypershift/hypershift-operator:latest'
            - name: HYPERSHIFT_ADDON_IMAGE_NAME
              value: 'quay.io/ianzhang366/hypershift-addon-operator:latest'
          imagePullPolicy: ""
          securityContext:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
              - ALL
            privileged: false
            readOnlyRootFilesystem: true
          resources:
            requests:
              cpu: "2m"
              memory: "2Mi"
---
# Source: hypershift-addon-chart/templates/hypershiftDeployment-deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    name: hypershift-deployment-controller
  name: hypershift-deployment-controller
  namespace: default
spec:
  replicas: 2
  selector:
    matchLabels:
      name: hypershift-deployment-controller
  template:
    metadata:
      labels:
        name: hypershift-deployment-controller
        ocm-antiaffinity-selector: hypershift-deployment-controller
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kubernetes.io/arch
                operator: In
                values:
                - amd64
                - ppc64le
                - s390x
                - arm64
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - weight: 70
            podAffinityTerm:
              topologyKey: topology.kubernetes.io/zone
              labelSelector:
                matchExpressions:
                - key: ocm-antiaffinity-selector
                  operator: In
                  values:
                  - hypershift-deployment-controller
          - weight: 35
            podAffinityTerm:
              topologyKey: kubernetes.io/hostname
              labelSelector:
                matchExpressions:
                - key: ocm-antiaffinity-selector
                  operator: In
                  values:
                  - hypershift-deployment-controller
      serviceAccountName: hypershift-deployment
      hostNetwork: false
      hostPID: false
      hostIPC: false
      securityContext:
        runAsNonRoot: true
      containers:
      - image: "quay.io/stolostron/hypershift-deployment-controller:latest"
        name: hypershift-deployment-controller
        command: ["./manager", "--leader-elect"]
        imagePullPolicy: ""
        env:
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        imagePullPolicy: ""
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - ALL
          privileged: false
          readOnlyRootFilesystem: true
        resources:
          requests:
            cpu: "2m"                     # Runs < 10m most of the time and idols at about 3m
            memory: "2Mi"                # Runs between 25-28Mi when fully active

```

Signed-off-by: Ian Zhang <izhang@redhat.com>

### CODE REVIEW

File: templates/hypershift-addon-manager-clusterrolebinding.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
   name: cluster-admin
 subjects:
   - kind: ServiceAccount
-    name: {{ .Values.hypershiftAddonMgr.serviceaccount }}
-    namespace: {{ .Values.installNamespace }}
\ No newline at end of file
+    name: {{ .Values.hypershiftAddonMgrOperator.serviceaccount }}
+    namespace: {{ .Values.org }}
```
Comment (zkayyali812) [inline templates/hypershift-addon-manager-clusterrolebinding.yaml]: This should be ` {{ .Values.global.namespace }}`

File: templates/hypershift-addon-manager-deployment.yaml [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
-  name: {{ .Values.hypershiftAddonMgr.deploymentName }}
-  namespace: {{ .Values.installNamespace }}
+  name: {{ .Values.hypershiftAddonMgrOperator.name }}
+  namespace: {{ .Values.global.namespace }}
   labels:
-    app: {{ .Values.hypershiftAddonMgr.deploymentName }}
+    app: {{ .Values.hypershiftAddonMgrOperator.name }}
 spec:
-  replicas: {{ .Values.hypershiftAddonMgr.replicaCount }}
+  replicas: 1
```
Comment (zkayyali812) [inline templates/hypershift-addon-manager-deployment.yaml:9]: Can you confirm that there should always be a replicaCount of 1?
Does this deployment have leaderelection, and can/should it support HA mode if configured in the MCH?

**Review by zkayyali812 — CHANGES_REQUESTED** [review]

File: templates/hypershift-addon-manager-deployment.yaml (hunk 2 above)
Comment (ianzhang366) [reply templates/hypershift-addon-manager-deployment.yaml:9]: we might be able to make it HA, but this relaase, we might not have cycle for it...in this case, what's your suggestion?

Comment (zkayyali812) [reply templates/hypershift-addon-manager-deployment.yaml:9]: That is ok, we can leave as 1 for now and update once leaderelection is validated.
 
Currently we prefer if components support our HA toggle, but this is not a hard requirement for us now as far as I know.

We can always get this updated quickly if/when changes are made to support HA

**Review by zkayyali812 — APPROVED** [review]

**Pull request merged**